asserts/model.go

```diff
@@ -25,16 +25,310 @@ import (
 	"strings"
 	"time"
 
+	"github.com/snapcore/snapd/snap/channel"
 	"github.com/snapcore/snapd/snap/naming"
 	"github.com/snapcore/snapd/strutil"
 )
 
+// ModelSnap holds the details about a snap specified by a model assertion.
+type ModelSnap struct {
+	Name   string
+	SnapID string
+	// SnapType is one of: app|base|gadget|kernel|core, default is app
+	SnapType string
```

Comment:
Is there a reason this is not of type `snap.Type` is it cyclic imports?

Reply:
asserts is used in places where we don't want all the bulk of the snap package, it doesn't make, see discussion with mvo above though

Reply:
Perhaps the simple string types could all move to `naming`?

Reply:
added TODO